libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -475,10 +475,8 @@ void AP_ESC_Telem::update_rpm(const uint8_t esc_index, const float new_rpm, cons
 
     rpmdata.prev_rpm = rpmdata.rpm;
     rpmdata.rpm = new_rpm;
-    if (now > last_update_us) { // cope with wrapping
-        rpmdata.update_rate_hz = 1.0e6f / (now - last_update_us);
-    }
-    rpmdata.last_update_us = now;
+    rpmdata.update_rate_hz = MAX(1.0e6f / (now - last_update_us), 0.1f); // clamp out incredibly slow updates
```

Comment:
I feel like the old code was generally more correct here. 0.1f is surely also wrong in most circumstances?

Reply:
If we look at coming back from an outage of say 20 seconds the old code would end up with `0.05` which works out to essentially mean the next computation of the RPM changing through the filter is almost nothing. The thought here was after *extreme* outage to try and make us do the initial update faster. (A poor man's filter reset). We could also just reset the filter entirely.

I'm totally happen to drop this part, but I do think it's better than the old behavior. What part did you like more?